src/main/java/gyro/google/storage/BucketAccessControlResource.java

```diff
@@ -0,0 +1,266 @@
+     * The associated Bucket.
+     */
+    @Required
+    public BucketResource getBucket() {
+        return bucket;
+    }
+
+    public void setBucket(BucketResource bucket) {
+        this.bucket = bucket;
+    }
+
+    /**
+     * The access permission for the entity. Valid values are ``OWNER``, ``READER``, or ``WRITER``.
+     */
+    @Required
+    @Updatable
+    @ValidStrings({ "OWNER", "READER", "WRITER" })
+    public String getRole() {
+        return role;
+    }
+
+    public void setRole(String role) {
+        this.role = role;
+    }
+
+    /**
+     * The entity holding the permission, in one of the following forms ``user-<userId>``, ``user-<email>``, ``group-<groupId>``, ``group-<email>``, ``domain-<domain>``, ``project-<team-projectId>``, ``allUsers``, or ``allAuthenticatedUsers``.
+     */
+    @Regex(ENTITY_REGEX)
+    @Required
+    @Updatable
+    public String getEntity() {
+        return entity;
+    }
+
+    public void setEntity(String entity) {
+        this.entity = entity;
+    }
+
+    /**
+     * The project to be billed for this request.
+     */
+    @Updatable
+    public String getUserProject() {
+        return userProject;
+    }
+
+    public void setUserProject(String userProject) {
+        this.userProject = userProject;
+    }
+
+    /**
+     * The domain associated with the entity.
+     */
+    @Output
+    public String getDomain() {
+        return domain;
+    }
+
+    public void setDomain(String domain) {
+        this.domain = domain;
+    }
+
+    /**
+     * Unique ID for the resource.
+     */
+    @Output
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * The link to this access-control entry.
+     */
+    @Output
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink;
+    }
+
+    /**
+     * The email address associated with the entity.
+     */
+    @Output
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    /**
+     * The ID for the entity.
+     */
+    @Output
+    public String getEntityId() {
+        return entityId;
+    }
+
+    public void setEntityId(String entityId) {
+        this.entityId = entityId;
+    }
+
+    @Override
+    public boolean doRefresh() throws Exception {
+        Storage storage = createClient(Storage.class);
+
+        try {
+            BucketAccessControl acl = storage.bucketAccessControls()
+                .get(getBucket().getName(), getEntity())
+                .setUserProject(getUserProject())
+                .execute();
+
+            if (acl == null) {
+                return false;
+            }
+
+            copyFrom(acl);
+
+            return true;
+        } catch (GoogleJsonResponseException e) {
+            if (e.getDetails().getCode() == 404) {
```

Comment:
This is caught by `GoogleResource.refresh()`.

Reply:
Fixed.